pkg/cli/commands.go

```diff
@@ -56,19 +56,23 @@ func MakeMainCommand[T field.Configurable](
 			return err
 		}
 
-		l := ctxzap.Extract(runCtx)
-
-		otelShutdown, err := uotel.InitOtel(context.Background(), v.GetString("otel-collector-endpoint"), name)
+		runCtx, otelShutdown, err := initOtel(runCtx, name, v, nil)
 		if err != nil {
 			return err
 		}
 		defer func() {
+			if otelShutdown == nil {
+				return
+			}
```

Comment:
Do we want to call otelShutdown with runCtx instead of context.Background()? I didn't actually look at what otelShutdown does.